src/Sentry/ScopeExtensions.cs

```diff
@@ -307,10 +338,21 @@ public static void AddEventProcessors(this Scope scope, IEnumerable<ISentryEvent
             AttachmentType type = AttachmentType.Default,
             string? contentType = null)
         {
-            var stream = File.OpenRead(filePath);
-            var fileName = Path.GetFileName(filePath);
+            try
+            {
+                var stream = File.OpenRead(filePath);
```

Comment:
Oh I just realized this.

So the stream stays open for the whole duration of the app?

In the other SDKs we read when the SDK is about to be sent. On Windows this will stop the file from being written to also.

Reply:
We should probably be specific about this on the docs: If you give a path, we'll read only when the event is going out.

Reply:
Yeah. That's the implication of streaming in envelopes.

Reply:
Well not for the duration of the app, but until the envelope is sent.

Reply:
Why can't we hold the path and open the team only if or when an event is sent?

Reply:
Lets discuss this before merging

Reply:
[Going into the docs btw](https://github.com/getsentry/sentry-docs/pull/2826/files#diff-624052c7e178a12fa0c1f439d8fd6913d3ae7bcac1cc3d939658f9f4cd0a874fR47-R50).

> The simplest way to create an attachment is to use a `path`. **The SDK is going to read the contents of the file each time** it is
preparing an event or transaction and is going to add the attachment to the same [envelope](https://develop.sentry.dev/sdk/envelopes/).
If the SDK can't read the file it logs an error message and drops the attachment.

Reply:
> Why can't we hold the path and open the team only if or when an event is sent?

Hm, I guess we can probably do that. However, it's worth noting a scenario that may happen in such case:

1. User adds `log.txt` as attachment. At the time of calling `AddAttachment` it had certain content. The operation was successful (file exists, everything is ok).
2. Before the envelope is sent, the application changes `log.txt` somehow so it's now technically a different file. Or even deleted.
3. The attachment sent to Sentry is different than the one the user initially "added".

This is a concern because the user needs to understand the internals of Sentry SDK, i.e. the fact that the envelopes are sent later and the fact that `AddAttachment` doesn't actually add anything at the moment of calling. This may lead to unexpected results.

On the other hand, locking the file (by holding the stream) obviously prevents that.

Reply:
okay, makes sense